components/omega/src/ocn/Eos.cpp

```diff
@@ -0,0 +1,152 @@
+#include "Eos.h"
+#include "DataTypes.h"
+#include "HorzMesh.h"
+
+namespace OMEGA {
+
+Eos *Eos::DefaultEos = nullptr;
+std::map<std::string, std::unique_ptr<Eos>> Eos::AllEos;
+
+TEOS10Poly75t::TEOS10Poly75t() {
+   specVolPcoeffs = Array2DReal("specVolPcoeffs", 6, VecLength);
+}
+
+LinearEOS::LinearEOS() {}
+
+Eos::Eos(const std::string &Name, ///< [in] Name for eos object
+         const HorzMesh *Mesh,    ///< [in] Horizontal mesh
+         int NVertLevels          ///< [in] Number of vertical levels
+) {
+   SpecVol = Array2DReal("SpecVol", Mesh->NCellsAll, NVertLevels);
+   SpecVolDisplaced =
+       Array2DReal("SpecVolDisplaced", Mesh->NCellsAll, NVertLevels);
```

Comment:
We should create Fields for these variables (so they can be used with IOStreams) in the constructor (see https://github.com/E3SM-Project/Omega/blob/85c270a98d35d5460dcb46d0b9fa63f749f20ea9/components/omega/src/ocn/OceanState.cpp#L185-L271) and destroy the fields in the destructor (https://github.com/E3SM-Project/Omega/blob/85c270a98d35d5460dcb46d0b9fa63f749f20ea9/components/omega/src/ocn/)OceanState.cpp#L149-L164

Reply:
Should these be in the Eos constructor? If so, are they their own Group? 
Or should Specific Volume be included in the ocean state?

Reply:
Also, should I define them similarly with both Host and Device arrays, e.g.
https://github.com/E3SM-Project/Omega/blob/85c270a98d35d5460dcb46d0b9fa63f749f20ea9/components/omega/src/ocn/OceanState.h#L85-L89